### DISCUSSION TITLE
How to I get TOC in vue to use in my custom theme ? params is undifined.

### DISCUSSION THREAD

**Discussion by jahidhsn002** [vuejs/vitepress#3306] [Q&A] [answered]
How to I get TOC in vue to use in my custom theme ? 
![Screenshot from 2023-12-08 11-35-51](https://github.com/vuejs/vitepress/assets/17622357/36874469-a899-4e22-bbc8-e411ad928f9b)
{ params } from useData() is undifined in custom theme ...

**Comment by brc-dd** [comment, marked as answer]
params is for dynamic routes parameters. you need page.value.headers. You need to enable this in config - https://github.com/vuejs/vitepress/issues/2840#issuecomment-1691062367

**Reply by jahidhsn002** [reply]
Thanks solved working....
on config.js .....
```
export default defineConfig({
  ........
  markdown: {
    headers: {
      level: [2,3,4]
    }
  },
})
```
